app/controllers/spree/checkout_controller_decorator.rb

```diff
@@ -30,8 +30,4 @@ def before_address
     @order.bill_address ||= preferred_bill_address || last_used_bill_address || Spree::Address.default
     @order.ship_address ||= preferred_ship_address || last_used_ship_address || nil
   end
-
-  def after_complete
-    reset_order
-  end
```

Comment:
This method is the culprit of the bug. I copy here the explanation I wrote in https://github.com/Em-AK/openfoodnetwork/pull/8/commits/291a9ffabed1dd02bae64410cae227ae011c7415

Spree does not call after_<order.state> methods any more as of
spree/spree#2557, so our `#after_complete` method
is never triggered and thus the order never reset.

This makes the condition:

```ruby
if current_order.andand.distributor == @order.distributor
```

in `app/views/spree/orders/form/_update_buttons.html.haml` return false
and as a result the "Back To Cart" button is not shown.

This commit resets the order (emptying the `session[:order_id]` and
creating a new order, aka. cart) right from the
`CheckoutController#update` rather than relying on infernal callbacks (of
what the Spree core team itself was unhappy about since long ago
spree/spree#2488). There is the first place
where we know the order has been successfully completed.